code_producers/src/llvm_elements/instructions.rs

```diff
@@ -860,3 +860,18 @@ pub fn create_switch<'a>(
 ) -> InstructionValue<'a> {
     producer.builder().build_switch(value, else_block, cases)
 }
+
+pub fn get_index_from_gep(gep: PointerValue) -> usize {
+    let op = gep
```

Comment:
Maybe it should assert that `gep` has the GEP opcode.

Reply:
Maybe also assert that operand 1 is `0`?

Reply:
That's exactly why I would add the assert. If some weird case ever appeared where it wasn't 0 for some reason, we would want to know.